### DISCUSSION TITLE
OIDC authentication redirects to `https` even if `provider_url` is `http`.

### DISCUSSION THREAD

**Discussion by deadnews** [semaphoreui/semaphore#1996] [Q&A] [answered]
## Description

OIDC authentication redirects to `https` even if `provider_url` is `http`.
The issue is not caused by the browser redirecting to `https`.
Is it `semaphore` that is redirecting to `https`?

## Steps to Reproduce

Here's the relevant configuration:

```json
  "oidc_providers": {
    "gitlab": {
      "display_name": "Sign in with GitLab",
      "color": "orange",
      "icon": "gitlab",
      "provider_url": "http://gitlab.domain.com",
      "client_id": "***",
      "client_secret": "gloas-***",
      "redirect_url": "http://semaphore.domain.com/api/auth/oidc/gitlab/redirect"
    }
  }
```

## Context

Semaphore version: `v2.9.75`
GitLab version: `16.11 CE`

**Comment by tboerger** [comment, marked as answer]
If I remember correctly it's part of the oidc specification to enforce https.
